### PR TITLE
percona-server: enable pam plugin

### PR DESCRIPTION
**Pull request by solomonty**
Fixes https://github.com/Homebrew/legacy-homebrew/issues/41789

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by solomonty** [comment]
There is another open PR for this formula. What is the process?

File: Formula/percona-server.rb [hunk 1]
```diff
@@ -72,13 +72,8 @@ def install
       -DCOMPILATION_COMMENT=Homebrew
       -DWITH_EDITLINE=system
       -DCMAKE_BUILD_TYPE=RelWithDebInfo
-    ]
-
-    # PAM plugin is Linux-only at the moment
```
Comment (fxcoudert) [inline Formula/percona-server.rb]: Can you, in the PR (not in the formula) give a link that confirms PAM works for macOS now? (release notes or something like that)

File: Formula/percona-server.rb [hunk 2]
```diff
@@ -72,13 +72,8 @@ def install
...
-    args.concat %w[
-      -DWITHOUT_AUTH_PAM=1
-      -DWITHOUT_AUTH_PAM_COMPAT=1
-      -DWITHOUT_DIALOG=1
+      -DWITH_PAM=yes
+      -DHAVE_STRUCT_IFREQ_IFR_NAME=yes
     ]
```
Comment (fxcoudert) [inline Formula/percona-server.rb]: Add a `revision 1` line at the end of the `homepage`/`url`/`sha256` block.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/percona-server.rb (hunk 1 above)
Comment (solomonty) [inline Formula/percona-server.rb]: The purpose of this PR is to add the `dialog.so` plugin so the `mysql` command line and other programs like `Sequel Pro` and `MySQLWorkbench` can use it.

```
MySQL said: Authentication plugin 'dialog' cannot be loaded: dlopen(/usr/local/lib/plugin/dialog.so, 2): image not found
```

The Percona PAM authentication plugins for MySQL are not built unless `WITH_PAM` is set.

File: Formula/percona-server.rb (hunk 2 above)
Comment (solomonty) [inline Formula/percona-server.rb]: Done

File: Formula/percona-server.rb [hunk 3]
```diff
@@ -55,6 +56,10 @@ def install
       "COMMAND /usr/bin/libtool -static -o ${TARGET_LOCATION}",
       "COMMAND libtool -static -o ${TARGET_LOCATION}"
 
+    inreplace "plugin/percona-pam-for-mysql/src/dialog.c",
+      "#include <stdlib.h>",
+      "#include <stdlib.h>\n#include <../strings/my_stpnmov.c>"
```
Comment (fxcoudert) [inline Formula/percona-server.rb]: What is the bug you are fixing? Has it been reported upstream and are they integrating the patch there? Add a comment giving link and saying that the `inreplace` can be removed for next version.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/percona-server.rb (hunk 3 above)
Comment (solomonty) [inline Formula/percona-server.rb]: I reported the bug to Percona.

https://bugs.launchpad.net/percona-server/+bug/1671357

**Comment by solomonty** [comment]
Is there a mechanism to force a rebuild in Jenkins? The sierra build failed trying to download boost.

```
Error: 1 problem in 1 formula
percona-server:
  * Stable resource "boost": The URL https://downloads.sourceforge.net/project/boost/boost/1.59.0/boost_1_59_0.tar.bz2 is not reachable
```

File: Formula/percona-server.rb (hunk 3 above)
Comment (fxcoudert) [inline Formula/percona-server.rb]: Add a comment giving link and saying that the inreplace can be removed for next version.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/percona-server.rb (hunk 3 above)
Comment (solomonty) [inline Formula/percona-server.rb]: Added the comments.

**Comment by solomonty** [comment]
Comments with links added

**Comment by fxcoudert** [comment]
So far I don't see movement from upstream on that bug report. Machomebrew only carries patches that have been approved or applied upstream. We'll wait to see if they act.

**Comment by solomonty** [comment]
I did not see that in the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) or in the [Formula Cookbook](http://docs.brew.sh/Formula-Cookbook.html). @DomT4 did request a PR to fix this in the referenced issue.

**Comment by solomonty** [comment]
rebased

**Pull request closed without merging**